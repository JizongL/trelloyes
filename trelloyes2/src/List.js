import React,{Component} from 'react'
import Card from './Card'
import './List.css'
import renderer from 'react-test-renderer'

const List = function(props){
  console.log(props.cards);
  return(
    <section className="List">
      <header>
      <h2>{props.header}</h2>
      </header>
      <div className='List-cards'>
        {props.cards.map((card,i) =>
          <Card
            
            key={i}
            title={card.title}
            content={card.content}
          />
        )}
      </div>
    </section>
  )
}


export default List;
