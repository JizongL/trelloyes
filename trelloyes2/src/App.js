import React, { Component } from 'react';
import List from './List'
import './App.css';

class App extends Component {
  static defaultProps = {
    store: {
      lists: [],
      allCards: {},
    }
  };

  render() {
    const{store} = this.props;
    
    
    
    return (
      
      <div className="App">
        {store.lists.map(
          list=> <List 
          header={list.header} 
          
          cards = {list.cardIds.map(id=>store.allCards[id])
          
          } 
          
          key={list.id}>
          </List>
        )}
      </div>
    );
  }
}

export default App;
