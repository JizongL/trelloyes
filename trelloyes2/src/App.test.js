import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import List from './List'
import Card from './Card'
import STORE from './store'
import renderer from 'react-test-renderer'


describe('My Components',()=>{
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  
  it('render List Component as expected',()=>{
    const tree = renderer
    .create(<List 
      header={STORE.lists[0].header} 
      
      cards = {STORE.lists[0].cardIds.map(id=>STORE.allCards[id])
      
      } 
      
      key={STORE.lists[0].id}>
      </List>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it('render Card Component as expected',()=>{
    const tree = renderer
    .create(<Card
              
      key={'a'}
      title={STORE.allCards['a'].title}
      content={STORE.allCards['a'].content}
      
    />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  })
  
})
