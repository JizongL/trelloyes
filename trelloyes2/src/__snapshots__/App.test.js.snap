// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Components render Card Component as expected 1`] = `
<div
  className="Card"
>
  <h3>
    First card
  </h3>
  <p>
    lorem ipsum
  </p>
</div>
`;

exports[`My Components render List Component as expected 1`] = `
<section
  className="List"
>
  <header>
    <h2>
      First list
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Second card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Fifth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Sixth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Seventh card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Tenth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Twelth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Thirteenth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
</section>
`;
